### PR TITLE
Fix DSN regexp

### LINKED ISSUE #32: ParseDSN fails on a special access_key
If an asscss_key like `oRdsomkey5hg=` in the `dsn`, the `ParseDSN` returns an error.
I think we should fix the `reDSN`

### PR DESCRIPTION
**Pull request by weiguoz**
fix #32

### CODE REVIEW

**Review by tonyyang-svail — APPROVED** [review]
LGTM!

File: dsn.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 
 var (
 	// Regexp syntax: https://github.com/google/re2/wiki/Syntax
-	reDSN   = regexp.MustCompile(`^([a-zA-Z0-9_-]+):([a-zA-Z0-9_-]+)@([a-zA-Z0-9/_.-]+)\?([^/]+)$`)
+	reDSN   = regexp.MustCompile(`^([a-zA-Z0-9_-]+):([=a-zA-Z0-9_-]+)@([:a-zA-Z0-9/_.-]+)\?([^/]+)$`)
```
Comment (typhoonzero) [inline dsn.go:11]: Seems endpoint part do not need `:`?

Comment (weiguoz) [reply dsn.go:11]: endpoints such as: `maxcompute.server.com:8080` or a test env `10.1.2.3:8002`

Comment (typhoonzero) [reply dsn.go:11]: I see!

**Pull request merged**